byway/src/main/webapp/routepage.js

```diff
@@ -69,60 +100,115 @@ function calcRoute(directionsService, directionsRenderer, start, end) {
       window.alert("Could not calculate route due to: " + status);
     }
   });
+  // add starting point back into schedule panel for better visuals
+
+  route.splice(0,0,start);
+  console.log(route);
+  renderRouteList();
 }
 
-/** Clear the stops panel in the html */
-function clearStops(){
-  const stopList = document.getElementById('stop-list');
-  if(stopList != null){
-    stopList.innerText = "";
+/**
+ * Calculates and sums up the distance and time duration between all destinations (legs)
+ * @param {response} result response from the directions service object
+ */
+function computeDistanceTime(result) {
+  let totalDist = 0;
+  let totalTime = 0;
+  // full route
+  let route_response = result.routes[0];
+  let waypoint_order = route_response.waypoint_order;
+  
+  let route_copy = [...route];
+  for(let i = 0; i < route.length; i++){
+    route[i] = route_copy[waypoint_order[i]];
+  }
+
+  for (let i = 0; i < route_response.legs.length; i++) {
+    // in meters
+    totalDist += route_response.legs[i].distance.value;
+    // in seconds
+    totalTime += route_response.legs[i].duration.value;
   }
+  
+  totalDist = (totalDist / 1000).toFixed(2);
+  let hours = Math.floor(totalTime / 3600);
+  let minutes = Math.round((totalTime - hours*3600) / 60);
+ 
+  updateDistanceTime(totalDist, hours, minutes);
 }
 
-/** Get the new list of stops from datastore onload */
-function getStopsOnload(){
-  clearStops();
+/**
+ * Updates total distance and duration elements in HTML
+ * @param {number} distance total driving distance for whole route
+ * @param {number} hours estimated driving time in hours
+ * @param {number} minutes estimated driving time in minutes
+ */
+function updateDistanceTime(distance, hours, minutes){
+  document.getElementById("distance").innerText = distance + "km";
+  document.getElementById("duration").innerText = 
+    (hours == 0) ? minutes + " mins" : hours + " hr " + minutes + " mins";
+}
+ 
+/** Clear the route panel in the html */
+function clearRoute(){
+  const routeList = document.getElementById('route-list');
+  if(routeList != null){
+    routeList.innerText = "";
+  }
+}
+ 
+/** Get trip info from datastore onload */
+function getRouteOnload(){
+  clearRoute();
   fetch('/api/stop')
   .then(response => response.json())
-  .then((stopsResponse) => {
-    if(stopsResponse != null){
-      stops.push(...stopsResponse);
+  .then((trip) => {
+    if(trip != null){
+      start = end = trip.start;
+      destinations = trip.destinations;
+      route.push(...trip.route)
+
+      calcRoute();
+      renderRouteList();
```

Comment:
Note that `calcRoute()` will call `updateRoute()` will call `renderRouteList()`, making this call redundant. This can be removed.

Reply:
Oh good catch, thanks! Changed.